src/domain/UBGraphicsScene.cpp

```diff
@@ -2052,6 +2055,27 @@ bool UBGraphicsScene::isEmpty() const
     return mItemCount == 0;
 }
 
+bool UBGraphicsScene::eventFilter(QObject *watched, QEvent *event)
```

Comment:
Is there a reason to use an `eventFilter` in the scene instead of overriding `event()` in the view? I would assume (but have not tested!) that the touch events arrive at the view and can be handled there. If this is the case, then I would also move the event handling to the view, while the multi-touch drawing code could be here in the scene.

What you're touching with your finger is a screen with a widget, which is the view. You don't "touch" the scene. The scene just holds all the graphics items.